File: ruby/bob/bob.rb
class Bob

  def hey(input)
    @input = input
    return 'Fine. Be that way!' if nothing?
    return 'Woah, chill out!' if shout?
    return 'Sure.' if question?
    "Whatever."
  end

  def nothing?
    @input.match(/\s+/).to_s == @input
  end

  def question?
    @input[-1] == '?'
  end

  def shout?
    return false unless @input.match(/[A-Z]/)
    !@input.match(/[a-z]/)
  end

end

File: javascript/bob/bob.js
var Bob = function() {
  var self = {};

  function nothing(input) {
    if (input === '') return true;
    var ma = input.match(/\s+/);
    return ma && (ma.toString() === input);
  }

  function question(input) {
    return input.substr(-1) === '?'
  }

  function shout(input) {
    if (!input.match(/[A-Z]/)) return false;
    return !input.match(/[a-zäöü]/)
  }

  function hey(input) {
    if (nothing(input))  return 'Fine. Be that way!';
    if (shout(input))    return 'Woah, chill out!';
    if (question(input)) return 'Sure.';
    return "Whatever."
  }

  self.hey = hey;

  return self;
}; 



module.exports = Bob;